### ISSUE TITLE
test_features fails inexplicably with pandas error

### ISSUE THREAD

**Issue by Dlux804** [Dlux804/McQuade-Chem-ML#46]
**Describe the bug**
The test_features.py function test_featurize() fails when calling features.featurize().  It fails with an index error from pandas that none of the other tests raise.  It has to do with dropping the "calculated" columns.  

**To Reproduce**
Branch: Feature_A

1. Go to branch: Feature_A
2. Run pytest on test_features.py
3. Scroll down to failed test

**Expected behavior**
features.featurize() should run just fine, not giving any errors.  Just like it does in previous tests in the same file. 

**Environment:**
Pycharm and CircleCI produce this.  

**Additional context**
Full Error: 
```
test_features.py::test_featurize FAILED                                  [ 75%]Previous PATH: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML
Now in: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML\dataFiles
You have selected the following featurizations:    rdkit2d
Calculating features... Done.
You have selected the following featurizations:    rdkit2d
Calculating features... Done.
Switching back to previous PATH: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML

test_features.py:63 (test_featurize)
def test_featurize():
        """
        This function was designed to test the script features.py and its function "featurize"
    
        """
        # change working directory to
        os.chdir(ROOT_DIR)
        # move to dataFiles
        with misc.cd('dataFiles'):
            print('Now in:', os.getcwd())
            # Read csv into dataframe
            df = pd.read_csv("water-energy.csv")
            # Next, test on every model. We know that the feats are different if the model is either nn or knn
            for i in model_name:
                # Exception with nn and knn
                if i == 'nn' or i == 'knn':
                    df, selected_feat, feat_time = features.featurize(df, i, num_feat=[0])  # Call function featurize
                    # Test to see if df is a dataframe
                    assert type(df) == pd.DataFrame, 'Something is wrong, i can feel it'
                    # Test to see if the variable time is a number
                    assert type(feat_time) == float, 'Something is wrong, i can feel it'
                else:
>                   df, selected_feat, feat_time = features.featurize(df, i, num_feat=[0])

tests\test_features.py:86: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
core\features.py:71: in featurize
    df = df[df.columns.drop(list(df.filter(regex='_calculated')))]
C:\Users\Adam\miniconda3\envs\mlapp\lib\site-packages\pandas\core\indexes\base.py:5013: in drop
    indexer = self.get_indexer(labels)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index(['iupac', 'smiles', 'expt', 'calc', 'BalabanJ', 'BertzCT', 'Chi0',
       'Chi0n', 'Chi0v', 'Chi1',
       ...
 ...iazole', 'fr_thiocyan', 'fr_thiophene', 'fr_unbrch_alkane',
       'fr_urea', 'qed'],
      dtype='object', length=405)
target = Index(['RDKit2D_calculated'], dtype='object'), method = None
limit = None, tolerance = None

    @Appender(_index_shared_docs["get_indexer"] % _index_doc_kwargs)
    def get_indexer(self, target, method=None, limit=None, tolerance=None):
        method = missing.clean_reindex_fill_method(method)
        target = ensure_index(target)
        if tolerance is not None:
            tolerance = self._convert_tolerance(tolerance, target)
    
        # Treat boolean labels passed to a numeric index as not found. Without
        # this fix False and True would be treated as 0 and 1 respectively.
        # (GH #16877)
        if target.is_boolean() and self.is_numeric():
            return ensure_platform_int(np.repeat(-1, target.size))
    
        pself, ptarget = self._maybe_promote(target)
        if pself is not self or ptarget is not target:
            return pself.get_indexer(
                ptarget, method=method, limit=limit, tolerance=tolerance
            )
    
        if not is_dtype_equal(self.dtype, target.dtype):
            this = self.astype(object)
            target = target.astype(object)
            return this.get_indexer(
                target, method=method, limit=limit, tolerance=tolerance
            )
    
        if not self.is_unique:
            raise InvalidIndexError(
>               "Reindexing only valid with uniquely valued Index objects"
            )
E           pandas.core.indexes.base.InvalidIndexError: Reindexing only valid with uniquely valued Index objects

C:\Users\Adam\miniconda3\envs\mlapp\lib\site-packages\pandas\core\indexes\base.py:2734: InvalidIndexError

test_features.py::test_targets_features PASSED                           [100%]Previous PATH: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML
Now in: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML\dataFiles
Switching back to previous PATH: C:\Users\Adam\OneDrive\Research\McQuade\Projects\NSF\OKN\phase1\Work\scripting\McQuade-Chem-ML

```

**Comment by qle2** [comment]
@Dlux804 
I tested your features.py script on my branch and it seems to run just fine on Pycharm but runs into an error with Circle CI like you observed.
![bug](https://user-images.githubusercontent.com/56049656/83051057-75415b00-a01b-11ea-9b8c-6d3d6d85c5d9.PNG)

I adjusted `df = df[df.columns.drop(list(df.filter(regex='_calculated')))]` to `df = df.drop(list(df.filter(regex='_calculated')), axis=1)` (they do the exact same thing) and it passes the Circle CI test.
![pass](https://user-images.githubusercontent.com/56049656/83051259-c05b6e00-a01b-11ea-9695-cf7ab9dd8c12.PNG)

**Comment by qle2** [comment]
Issue resolved in Pull request #44
